DQM/TrackingMonitorSource/python/PPonPbPbTrackingOnly_custom.py

```diff
@@ -0,0 +1,17 @@
+"""
+_Utils_
```

Comment:
Why do we need this file at all?
Seems a straight copy of 
https://github.com/cms-sw/cmssw/blob/30a3bf42129af75e7cd79b2bc3b55dcfb9e8d862/DQM/TrackingMonitorSource/python/PPonAATrackingOnly_custom.py#L1-L18

Reply:
My first attempt was to use this file PPonAATrackingOnly, but I got a 'ProductNotFound' crash when running path `dqmoffline_step`. So my poor (wo)man's solution was to replace `dqmofflineOnPAT_step`. :-)
I am more than happy to take any suggestion on how to handle it in a smarter way.
Just for the record, this was the crash I got when using [PPonAATrackingOnly_custom.py](https://github.com/cms-sw/cmssw/blob/30a3bf42129af75e7cd79b2bc3b55dcfb9e8d862/DQM/TrackingMonitorSource/python/PPonAATrackingOnly_custom.py#L1-L18).

`
----- Begin Fatal Exception 12-Sep-2022 18:11:30 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 325174 lumi: 1 event: 32100 stream: 0
   [1] Running path 'dqmoffline_step'
   [2] Calling method for module TrackSelector/'hiConformalPixelTracksQA'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: std::vector<reco::Track>
Looking for module label: hiConformalPixelTracks
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
`

Reply:
I think the problem might originate from the fact that in your test command

```console
python3 Configuration/DataProcessing/test/RunExpressProcessing.py --scenario trackingOnlyEra_Run3_pp_on_PbPb --global-tag 124X_dataRun3_Express_TIER0_REPLAY_Run2_v2 --lfn file:write_PrimDS1_RAW.root --alcarecos=TkAlMinBias+PromptCalibProdBeamSpotHP
```

you are not specifying the DQM sequence to be run.
If I look to the original configuration for that run [327527](https://cmsweb.cern.ch/t0wmadatasvc/prod/express_config?run=327527) the DQM sequence is configured to be:

```
  "dqm_seq": "DQMOfflineTracking"
```

conversely I am not sure how can the `RunExpressProcessing` be configured to accept something other then the default  since the options are:

```
RunExpressProcessing.py <options>

Where options are:
 --scenario=ScenarioName
 --raw (to enable RAW output)
 --reco (to enable RECO output)
 --fevt (to enable FEVT output)
 --dqm (to enable DQM output)
 --no-output (create config with no output, overrides other settings)
 --global-tag=GlobalTag
 --lfn=/store/input/lfn
 --alcarecos=plus_seprated_list
 --nThreads=Number_of_cores_or_Threads_used
```

There was an attempt earlier to allow to the ConfigDP machinery to use non-default express DQM sequences (see e.g. https://github.com/cms-sw/cmssw/pull/35850 ).
Finally this begs the question how can Tier-0 configure something which is not the default for an express-like stream, if the cmssw tools cannot provide it...

Reply:
incidentally, you would get the *very same error*  also when running with the 2018 era [1]:

```
python3 Configuration/DataProcessing/test/RunExpressProcessing.py --scenario trackingOnlyEra_Run2_2018_pp_on_AA --global-tag 124X_dataRun3_Express_TIER0_REPLAY_Run2_v2 --lfn file:write_PrimDS1_RAW.root --alcarecos=TkAlMinBias+PromptCalibProdBeamSpotHP
```
demonstrating there is no need for such new file.

[1]
```console
---- Begin Fatal Exception 14-Sep-2022 10:28:38 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 327527 lumi: 51 event: 19480105 stream: 0
   [1] Running path 'dqmoffline_step'
   [2] Calling method for module TrackSelector/'hiConformalPixelTracksQA'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: std::vector<reco::Track>
Looking for module label: hiConformalPixelTracks
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------

```

Reply:
OK, checking in the native release it actually works:

```console
cmsrel CMSSW_10_3_1
cd CMSSW_10_3_1/src/
git cms-addpkg Configuration/DataProcessing
scramv1 b -j 20
cmsenv
python Configuration/DataProcessing/test/RunRepack.py --lfn file:/eos/cms/store/t0streamer/Data/HIExpressAlignment/000/327/527/run327527_ls0051_streamHIExpressAlignment_StorageManager.dat
cmsRun -e RunRepackCfg.py
python Configuration/DataProcessing/test/RunExpressProcessing.py --scenario trackingOnlyEra_Run2_2018_pp_on_AA --global-tag 103X_dataRun2_Express_v2 --lfn file:write_PrimDS1_RAW.root --alcarecos=TkAlMinBias+PromptCalibProdBeamSpotHP
cmsRun -e RunExpressProcessingCfg.py
```

so something must have changed from 10_3_1 to 12_6_X in how the sequences are configured.

Reply:
Indeed the sequence was changed:

   * master: https://github.com/cms-sw/cmssw/blob/366658ea646312c7478ec2d419bcf65f8b10ca2c/DQMOffline/Configuration/python/DQMOffline_cff.py#L108-L110
   * 10_3_1: https://github.com/cms-sw/cmssw/blob/e26e084ca13ba0cdb7ea386d170abdfa8fe1de97/DQMOffline/Configuration/python/DQMOffline_cff.py#L112-L115
   
   
 my suggestion (to recover the functionality of the old scenario, which is currently broken as per https://github.com/cms-sw/cmssw/pull/39387#discussion_r970497256, would be to apply the following patch to this PR:
 
<details>
  <summary> PATCH </summary>
  
 
 ```diff
 diff --git a/Configuration/DataProcessing/python/RecoTLR.py b/Configuration/DataProcessing/python/RecoTLR.py
index 1c1ec2bf7ff..a86f874ae50 100644
--- a/Configuration/DataProcessing/python/RecoTLR.py
+++ b/Configuration/DataProcessing/python/RecoTLR.py
@@ -97,9 +97,7 @@ def customisePostEra_Run3_express_trackingOnly(process):
 
 def customisePostEra_Run3_pp_on_PbPb_express_trackingOnly(process):
     #start with repeat of 2018
-    #customisePostEra_Run2_2018_pp_on_AA_express_trackingOnly(process)
-    from DQM.TrackingMonitorSource.PPonPbPbTrackingOnly_custom import customise_PPonPbPbTrackingOnlyDQM as _customise_PPonPbPbTrackingOnlyDQM
-    _customise_PPonPbPbTrackingOnlyDQM(process)
+    customisePostEra_Run2_2018_pp_on_AA_express_trackingOnly(process)
     return process
 
 def customisePostEra_Run3_pp_on_PbPb(process):
diff --git a/DQM/TrackingMonitorSource/python/PPonAATrackingOnly_custom.py b/DQM/TrackingMonitorSource/python/PPonAATrackingOnly_custom.py
index e158a258313..8b34605b935 100644
--- a/DQM/TrackingMonitorSource/python/PPonAATrackingOnly_custom.py
+++ b/DQM/TrackingMonitorSource/python/PPonAATrackingOnly_custom.py
@@ -6,12 +6,14 @@ Tools to customise the DQM offline configuration run on the dedicated express-li
 import FWCore.ParameterSet.Config as cms
 
 def customise_PPonAATrackingOnlyDQM(process):
-    if hasattr(process,'dqmofflineOnPAT_step'):
+    if (hasattr(process,'dqmofflineOnPAT_step') or hasattr(process,'dqmoffline_step')):
         process=customise_DQMSequenceHiConformalTracks(process)
     return process   
 
 def customise_DQMSequenceHiConformalTracks(process):
     process.TrackingDQMSourceTier0Common.remove(process.hiConformalPixelTracksQA)      
+    process.TrackingDQMSourceTier0MinBias.remove(process.hiConformalPixelTracksQA)
+    process.TrackingDQMSourceTier0.remove(process.hiConformalPixelTracksQA)
     return process
 
 
diff --git a/DQM/TrackingMonitorSource/python/PPonPbPbTrackingOnly_custom.py b/DQM/TrackingMonitorSource/python/PPonPbPbTrackingOnly_custom.py
deleted file mode 100644
index 4044179f519..00000000000
--- a/DQM/TrackingMonitorSource/python/PPonPbPbTrackingOnly_custom.py
+++ /dev/null
@@ -1,17 +0,0 @@
-"""
-_Utils_
-Tools to customise the DQM offline configuration run on the dedicated express-like stream during pp_on_PbPb
-"""
-
-import FWCore.ParameterSet.Config as cms
-
-def customise_PPonPbPbTrackingOnlyDQM(process):
-    if hasattr(process,'dqmoffline_step'):
-        process=customise_DQMSequenceHiConformalTracks(process)
-    return process
-
-def customise_DQMSequenceHiConformalTracks(process):
-    process.dqmoffline_step.remove(process.hiConformalPixelTracksQA)
-    return process
-
-
```
</details>
I tested explicitly that it works for both 2018

```bash
python3 Configuration/DataProcessing/test/RunExpressProcessing.py --scenario trackingOnlyEra_Run2_2018_pp_on_AA --global-tag 124X_dataRun3_Express_TIER0_REPLAY_Run2_v2 --lfn file:write_PrimDS1_RAW.root --alcarecos=TkAlMinBias+PromptCalibProdBeamSpotHP
```

and Run-3

```bash
python3 Configuration/DataProcessing/test/RunExpressProcessing.py --scenario trackingOnlyEra_Run3_pp_on_PbPb --global-tag 124X_dataRun3_Express_TIER0_REPLAY_Run2_v2 --lfn file:write_PrimDS1_RAW.root --alcarecos=TkAlMinBias+PromptCalibProdBeamSpotHP

```

Reply:
Thanks a lot, Marco! I will include your patch in this PR.